ldms/src/sampler/cxi_sampler/cxi_sampler.c

```diff
@@ -578,10 +574,63 @@ static int constructor(ldmsd_plug_handle_t handle)
 
 static void destructor(ldmsd_plug_handle_t handle)
 {
+    cxi_t cxi = ldmsd_plug_ctxt_get(handle);
+    if (!cxi)
+        return;
+
+    // handle ldms_set_t deletion
+    if (cxi->set)
+        base_set_delete(cxi->base);
+
+    if (cxi->schema)
+        base_schema_delete(cxi->base);
+
+    // Free interface names
+    if (cxi->iface_names) {
+        for (int i = 0; i < cxi->iface_count; i++) {
+            free(cxi->iface_names[i]);
+        }
+        free(cxi->iface_names);
+    }
+
+    // Free tel_path and rh_path
+    free(cxi->tel_path);
+    free(cxi->rh_path);
+
+    // Free telemetry files
+    if (cxi->tel_files) {
+        for (int i = 0; i < cxi->iface_count; i++) {
+            for (int j = 0; j < cxi->tel_files[i].count; j++) {
+                free(cxi->tel_files[i].names[j]);
+            }
+        }
+        free(cxi->tel_files);
+    }
+
+    // Free retry handler files
+    if (cxi->rh_files) {
+        for (int i = 0; i < cxi->iface_count; i++) {
+            for (int j = 0; j < cxi->rh_files[i].count; j++) {
+                free(cxi->rh_files[i].names[j]);
+            }
+        }
+        free(cxi->rh_files);
+    }
+
+    // Free LDMS records
+    if (cxi->tel_rec)
+        ldms_record_delete(cxi->tel_rec);
+    if (cxi->rh_rec)
+        ldms_record_delete(cxi->rh_rec);
+
+    // Free base data
+    base_del(cxi->base);
+
+    // Finally, free the main structure
+    free(cxi);
 }
 
 static struct ldmsd_sampler cxi_plugin = {
```

Comment:
Just FYI, if you want to make this compatible with the latest API changes (which is probably why you removed .base.name), you will also need to change this struct definition to:

```struct ldmsd_sampler ldmsd_plugin_interface = {```

And then delete the get_plugin() function below.

Reply:
@morrone that has all been done in cxi_sampler ... and actually tested.